### ISSUE TITLE
Generate a test dataset without groundtruth

### ISSUE THREAD

**Issue by glennliu** [ShunChengWu/3DSSG#22]
Hi 

I read the previous issue about test dataset. #4 
I already have 'inseg.ply' generated for my data. Is there any method to generate data without a ground-truth label? 

Thanks

**Comment by glennliu** [comment]
I finally build a testset myself. I modified large parts of 'dataset_SGFN.py'.
- The segments are filtered based on groundtruth class type. And it build neighbor graph based on groundtruth indices. I removed large parts of that code. 
- metrics are calculated using gt label. Removed that as well. 

It overall runs well on 3RScan datatset with the pre-trained model. 
But when I tried segments.ply build in other dataset, such as SceneNN. The semantic label is not so good (semantic segmenation attached). Many objects are under-segmented. I think trainning a  new model is necessary for other dataset.
![Screenshot from 2022-01-23 19-31-16](https://user-images.githubusercontent.com/8009679/150676605-920270d3-adcd-4498-8918-4ad3f9ef0aae.png)
![Screenshot from 2022-01-23 19-40-51](https://user-images.githubusercontent.com/8009679/150676771-dde95770-1bad-42e5-89bc-f18e63ee7ac6.png)

![Screenshot from 2022-01-23 19-30-40](https://user-images.githubusercontent.com/8009679/150676780-55636847-938b-45de-ab06-b51ba765aa8f.png)


![Screenshot from 2022-01-23 19-40-20](https://user-images.githubusercontent.com/8009679/150676683-d4323fc5-f66d-44c0-950b-c687220b1b75.png)


Overall, it is nice work!

**Comment by ShunChengWu** [comment]
The domain soft should not be that significant. Make sure the geometry is with the same coordinate system as 3RScan. 
Also, make sure the edge generation is correct. You can see that the vertical wall is predicted as flat classes (e.g. sofa). This may indicate that you have the wrong coordinate system.

We have tested the incremental version of this method on a real-world scan with RealSense D435. This should have a larger domain shift than changing from one more complete scan to another. You can check this video
https://www.youtube.com/watch?v=lJUfezPmn7M

**Comment by glennliu** [comment]
Hi @ShunChengWu 

I checked the coordinate issue. The SceneNN should use the same coordinate system as 3RScan. Both of them use "Z-up" right-hand coordinate. I verified it by rendering one of these two datasets in the same coordinate. Both scans have their z-axis and horizontal plane aligned (based on my visual verifying).  This should be able to verify the coordinate issue.

Then, I created three "inseg.ply" for scan 096,223, and 231 from SceneNN. The inputs are based full reconstructed scene. I construct geometric segmentation by graph-cut clustering method. It is slower compare to Tateno's method, but invariant to depth noise and occlusion. The input geometric segments and semantic segments are showed below, 

![scene223_inseg](https://user-images.githubusercontent.com/8009679/151289803-f0cc9887-087a-46a1-94ed-758ab428a380.png)

![scene223_semantic](https://user-images.githubusercontent.com/8009679/151289815-0bb4cd8e-2c72-483a-bd71-b91b1499cb9f.png)

![scene096_inseg](https://user-images.githubusercontent.com/8009679/151289830-f2550368-e4e8-4e5f-81d8-1a032579fa34.png)

![scene096_semantic](https://user-images.githubusercontent.com/8009679/151289838-f7c49580-4e2a-4e59-a1bc-d36ede143657.png)

![scene231_inseg](https://user-images.githubusercontent.com/8009679/151289847-bc76e870-493b-4ee5-afd7-30187f2fca00.png)

![scene231_semantic](https://user-images.githubusercontent.com/8009679/151289851-792f7baf-0246-42ab-af08-9ab8cd4ac4ad.png)
![semantic_legend](https://user-images.githubusercontent.com/8009679/151289912-a7e8e596-3d0a-4435-9711-559c01d4e19b.png)


The good part is that wall and floor are mostly correct. Some parts of the wall can be wrongly labeled as window or curtain. But I think that's understandable.   
However, some of the other larger-sized elements are labeled wrong: doors, cabinet, bed and counter. The door element has horizontal/verticle conflicts you mentioned. 
And more frequently, smaller-sized instance are significantly labeled wrong. Many objects (TV, table parts, bed parts, cabinet parts) are labeled as chairs. 

Do you think have any suggestions to improve the results? 
In case you need better visualize the resutls, the ply files are summarized here. 

Thanks

**Comment by ShunChengWu** [comment]
Are you using the pre-trained model in your experiment? The segmentation is quite different than the one from inseg. You may need to retrain the network with your segmentation. 
Also, you can adjust the parameter of generating the neighbor graph. it seems like you don' consider relationships that much. You can reduce the receptive field of the neighbor graph to only consider a smaller range of segments nearby.

**Comment by glennliu** [comment]
> Are you using the pre-trained model in your experiment? The segmentation is quite different than the one from inseg. You may need to retrain the network with your segmentation. Also, you can adjust the parameter of generating the neighbor graph. it seems like you don' consider relationships that much. You can reduce the receptive field of the neighbor graph to only consider a smaller range of segments nearby.

Yes, I'm using the pre-trained model. And my segmentation result is denser than inseg.ply. 
My priority is segmentation accuracy. I did not consider inter-instance relationships that much. But the "same part" relationship label is important for me, as it merges the over-segmented instance. 

Thanks for your advice!

**Comment by ShunChengWu** [comment]
Although without proper evaluation, we also trained our method with super-voxel segmentation [1](https://github.com/MarkusEich/cpf_segmentation). With proper training, our method works fine. 
As long as you output the file to the acceptable format as inseg.ply, you can feed them to the data generation script `gen_data.py`. This script will give you only the "same part" relationship when no other information is provided.
